src/Installers/Windows/WindowsHostingBundle/Bundle.wxs

```diff
@@ -57,13 +57,126 @@
+
+        <util:RegistrySearch Id="opt_no_ancm_should_be_set" 
+                         Condition="NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)"
+                         Root="HKLM" 
+                         Key="Software\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)" 
+                         Value="OPT_NO_ANCM" 
+                         Result="exists" 
+                         Variable="OPT_NO_ANCM_Should_Be_Set"/>
+
+        <util:RegistrySearch Id="opt_no_fts_should_be_set" 
+                         Condition="NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)"
+                         After="opt_no_ancm_should_be_set"
+                         Root="HKLM" 
+                         Key="Software\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)" 
+                         Value="OPT_NO_FTS" 
+                         Result="exists" 
+                         Variable="OPT_NO_FTS_Should_Be_Set"/>
+
+        <util:RegistrySearch Id="opt_no_lts_should_be_set" 
+                         Condition="NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)"
+                         After="opt_no_fts_should_be_set"
+                         Root="HKLM" 
+                         Key="Software\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)" 
+                         Value="OPT_NO_LTS" 
+                         Result="exists" 
+                         Variable="OPT_NO_LTS_Should_Be_Set"/>
+
+        <util:RegistrySearch Id="opt_no_sharedfx_should_be_set" 
+                         Condition="NOT (OPT_NO_ANCM OR OPT_NO_ANCM=0 OR OPT_NO_FTS OR OPT_NO_FTS=0 OR OPT_NO_LTS OR OPT_NO_LTS=0 OR OPT_NO_SHAREDFX OR OPT_NO_SHAREDFX=0 OR OPT_NO_RUNTIME OR OPT_NO_RUNTIME=0 OR OPT_NO_X86 OR OPT_NO_X86=0 OR OPT_NO_SHARED_CONFIG_CHECK OR OPT_NO_SHARED_CONFIG_CHECK=0)"
+                         After="opt_no_lts_should_be_set"
+                         Root="HKLM" 
+                         Key="Software\Microsoft\dotnet\host\options\$(var.MajorVersion).$(var.MinorVersion)" 
+                         Value="OPT_NO_SHAREDFX" 
+                         Result="exists" 
+                         Variable="OPT_NO_SHAREDFX_Should_Be_Set"/>
+
+        <util:RegistrySearch Id="opt_no_runtime_should_be_set" 
```

Comment:
I think you can simplify this further using a preprocessor variable to hold the condition (since it's the same) and then using a foreach to generate a parameterized search, e.g.

```
<?define Options=OPT_NO_X86;OPT_NO_RUNTIME?>
    <?foreach Option in $(var.Options?>
    <util:RegistrySearch Id="$(var.Option)_should_be_set">
      
    </util:RegistrySearch>
    <?endforeach?>
```